src/storage/v2/inmemory/storage.cpp

```diff
@@ -1220,6 +1210,37 @@ Transaction InMemoryStorage::CreateTransaction(IsolationLevel isolation_level, S
   return {transaction_id, start_timestamp, isolation_level, storage_mode, false};
 }
 
+/// Main lock is taken by the caller.
+void InMemoryStorage::SetStorageMode(StorageMode storage_mode) {
+  std::unique_lock main_guard{main_lock_};
+  MG_ASSERT(
+      (storage_mode_ == StorageMode::IN_MEMORY_ANALYTICAL || storage_mode_ == StorageMode::IN_MEMORY_TRANSACTIONAL) &&
+      (storage_mode == StorageMode::IN_MEMORY_ANALYTICAL || storage_mode == StorageMode::IN_MEMORY_TRANSACTIONAL));
+  if (storage_mode_ != storage_mode) {
+    if (storage_mode == StorageMode::IN_MEMORY_ANALYTICAL) {
+      snapshot_runner_.Stop();
+    } else {
+      snapshot_runner_.Run("Snapshot", config_.durability.snapshot_interval, [this] {
+        auto const &repl_state = repl_state_;
+        if (auto maybe_error = this->CreateSnapshot(repl_state, {true}); maybe_error.HasError()) {
```

Comment:
Maybe an issue soon, repl_state will not exist inside storage, needs to be passed in to the `SetStorageMode` call

Reply:
Shouldn't that then be fixed in PR which will introduce this change? Otherwise I don't know how to access repl_state in the callee of the `SetStorageMode`?

Reply:
Two PR in flight at the same time....one of us will need to fix it :)